integration-tests/oidc/src/main/java/io/quarkus/it/keycloak/VertxResource.java

```diff
@@ -5,25 +5,22 @@
 
 import io.quarkus.vertx.http.runtime.security.QuarkusHttpUser;
 import io.vertx.core.Handler;
-import io.vertx.core.buffer.Buffer;
 import io.vertx.ext.web.Router;
 import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.BodyHandler;
 
 @ApplicationScoped
 public class VertxResource {
 
     void setup(@Observes Router router) {
-        router.route("/vertx").handler(new Handler<RoutingContext>() {
-            @Override
-            public void handle(RoutingContext event) {
-                event.request().bodyHandler(new Handler<Buffer>() {
+        router.route("/vertx")
+                .handler(BodyHandler.create())
+                .handler(new Handler<RoutingContext>() {
                     @Override
-                    public void handle(Buffer data) {
-                        event.response().end(data);
+                    public void handle(RoutingContext event) {
+                        event.end(event.body().asString());
```

Comment:
Isn't it less efficient to transform the body to a string?

Reply:
@cescoffier do you think it's worth making it `buffer()` or should I merge as is?

Reply:
Fixed.